tests/src/test/java/alluxio/shell/command/CpCommandTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.shell.command;
+
+import alluxio.AlluxioURI;
+import alluxio.client.ReadType;
+import alluxio.client.file.FileInStream;
+import alluxio.client.file.URIStatus;
+import alluxio.client.file.options.OpenFileOptions;
+import alluxio.exception.AlluxioException;
+import alluxio.shell.AbstractAlluxioShellTest;
+import alluxio.shell.AlluxioShellUtilsTest;
+
+import com.google.common.io.Closer;
+import org.apache.commons.io.IOUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+/**
+ * Tests for cp command.
+ */
+public final class CpCommandTest extends AbstractAlluxioShellTest {
+
+  /**
+   * Tests recursively copying a directory.
+   */
+  @Test
+  public void copyDirTest() throws Exception {
```

Comment:
this tests copy dir when dst doesn't exist.
could you add a test to do copy when (1) dst exists and is a file (expecting a failure), and (2) dst exists and is a dir (expect src to become a subdir in dst)